csharp/ql/lib/semmle/code/csharp/frameworks/ServiceStack.qll

```diff
@@ -0,0 +1,326 @@
+/**
+ * General modelling of ServiceStack framework including separate modules for:
+ *  - flow sources
+ *  - SQLi sinks
+ *  - XSS sinks
+ */
+
+import csharp
+private import semmle.code.csharp.dataflow.ExternalFlow
+
+/** A class representing a Service */
+private class ServiceClass extends Class {
+  ServiceClass() {
+    this.getBaseClass+().hasQualifiedName("ServiceStack", "Service") or
+    this.getABaseInterface+().hasQualifiedName("ServiceStack", "IService")
```

Comment:
Should this be `this.getABaseType*().getABaseInterface().hasQualifiedName("ServiceStack", "IService")`? That will then include sub classes of classes that implement the interface.